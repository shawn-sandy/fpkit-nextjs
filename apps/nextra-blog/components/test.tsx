import { getPagesUnderRoute, getAllPages } from "nextra/context";
import { PageList } from "@fpkit/nextjs";
import { PostsList } from "@fpkit/nextjs";
export const Test = ({ frontMatter }) => {
  const pages = getAllPages();
  console.log({ pages });
  return <PostsList postList={getAllPages()} showDescription />;
};

export default Test;
