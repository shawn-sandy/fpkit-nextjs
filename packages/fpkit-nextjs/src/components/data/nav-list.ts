
export const navListItems = [
    {
        kind: "MdxPage",
        name: "articles",
        route: "/articles",
        frontMatter: {
            type: "page",
            title: "Articles",
            date: "2020-01-01T00:00:00.000Z",
        },
    },
    {
        kind: "MdxPage",
        name: "index",
        route: "/",
        frontMatter: {
            type: "page",
            title: "About",
            date: "2020-01-01T00:00:00.000Z",
        },
    },
    {
        kind: "MdxPage",
        name: "posts",
        route: "/posts",
        frontMatter: {
            type: "page",
            title: "Posts",
            date: "2020-01-01T00:00:00.000Z",
        },
    },
];
