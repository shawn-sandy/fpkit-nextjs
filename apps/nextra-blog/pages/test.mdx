---
title: "Test component"
description: "Test component"
date: 2023-10-01
type: "page"
---

import { Test } from "../components/Test";

<Test />
