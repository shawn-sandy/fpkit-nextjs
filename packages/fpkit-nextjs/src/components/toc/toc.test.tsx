import React from "react";

import { describe, expect, test } from "vitest";
import { render, screen } from "@testing-library/react";
import { axe } from "vitest-axe";

import { headings } from "./toc.test.data";

// ðŸ‘‰ import ComponentName from "./component-name";
import Toc from "./toc";
describe("App renders correctly", () => {
  test("renders without crashing", async () => {
    render(<Toc items={headings} />);
    const toc = screen.getByRole("list");
    expect(toc).toBeDefined();
    expect(await axe(toc)).toHaveNoViolations();
    expect(toc).toMatchSnapshot();
  });
});
