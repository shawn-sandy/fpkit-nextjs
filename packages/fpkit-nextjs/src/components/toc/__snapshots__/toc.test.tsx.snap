// Vitest Snapshot v1

exports[`App renders correctly > renders without crashing 1`] = `
<ul
  data-style="true"
>
  <li
    data-list="unstyled"
  >
    <a
      href="#test-filename-line-highlighting-and-empty-lines"
    >
      Test filename, line highlighting and empty lines
    </a>
  </li>
  <li
    data-list="unstyled"
  >
    <a
      href="#test-showlinenumbers-and-word-highlighting"
    >
      Test showLineNumbers and word highlighting
    </a>
  </li>
  <li
    data-list="unstyled"
  >
    <a
      href="#test-highlighting-inline-code"
    >
      Test highlighting inline code
    </a>
  </li>
  <li
    data-list="unstyled"
  >
    <a
      href="#test-without-specified-language"
    >
      Test without specified language
    </a>
  </li>
  <li
    data-list="unstyled"
  >
    <a
      href="#test-link-in-code"
    >
      Test link in code
    </a>
  </li>
</ul>
`;
